src/libopensc/card-jpki.c

```diff
@@ -265,6 +271,7 @@ jpki_pin_cmd(sc_card_t * card, struct sc_pin_cmd_data *data, int *tries_left)
 		if (tries_left) {
 			*tries_left = apdu.sw2 - 0xC0;
 		}
+		data->pin1.logged_in = priv->state;
```

Comment:
In ISO-7816 `00 20, 00, 80` should return `90 00` if the user is logged in. Does your card return such information?